### PR TITLE
Release fix 182

### PR DESCRIPTION
**Pull request by Tricky12321**
This fixed issue #182

### CODE REVIEW

**Comment by User12837985123** [comment]
@aau-giraf/sw610f19 you have been assigned this. Here's your checklist. I marked @Moesgaarda as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [x] Is the code in the right place? (Both in terms of folder structure and class structure)
- [x] Could this code, to the best of your knowledge, have reused existing code?
- [x] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [x] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [x] Are names meaningful and self-documenting?
- [x] Is it understandable, by reading the code, what it does?
- [x] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [x] Is it understandable, by reading the tests, what they do?
- [x] Do the tests cover sensible cases? Both happy paths and exception paths?
- [x] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [x] Are error messages in the code understandable?
- [x] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [x] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [x] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [x] Are UI messages, that the user might meet, actually helpful?
- [x] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [x] Is there well written code in this? Remember to tell the developer!

**Comment by User12837985123** [comment]
@aau-giraf/sw609f19 you have been assigned this. Here's your checklist. I marked @MStenkaer as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

File: lib/blocs/weekplan_bloc.dart [hunk 1]
```diff
@@ -49,6 +53,7 @@ class WeekplanBloc extends BlocBase {
         .listen((WeekModel newWeek) {
       _userWeek.add(UserWeekModel(newWeek, user));
     });
+    _userWeek.add(UserWeekModel(week, user));
```
Comment (MStenkaer) [inline lib/blocs/weekplan_bloc.dart]: Why is this added?

**Review by MStenkaer — CHANGES_REQUESTED** [review]
There seems to be a bug:
1. Mark an activity as marked.
2. Leave the weekplan.
3. Reopen the weekplan, and the activity is unchecked again.

Also all of the used dependencies has been changed to older versions in pubspec.

File: test/blocs/weekplan_bloc_test.dart [hunk 2]
```diff
@@ -49,60 +103,60 @@ void main() {
...
         order: null,
         isChoiceBoard: null,
         state: null,
         id: null,
         pictogram: null);
-
+    weekplanBloc.loadWeek(week, user);
     weekplanBloc.userWeek.skip(1).listen((UserWeekModel userWeek) {
-      verify(api.week.update(any, any, any, any));
+      //verify(api.week.update(any, any, any, any));
```
Comment (jenspetur94) [inline test/blocs/weekplan_bloc_test.dart]: Code shouldn't be commented out. It should be fixed or removed

**Review by jenspetur94 — CHANGES_REQUESTED** [review]
we left a small comment with some commented out code.
when this and the bug is fixed we will approve

**Review by jenspetur94 — DISMISSED** [review]

File: lib/blocs/weekplan_bloc.dart [hunk 3]
```diff
@@ -49,6 +58,7 @@ class WeekplanBloc extends BlocBase {
...
```
Comment (MStenkaer) [inline lib/blocs/weekplan_bloc.dart]: Now it is back

Comment (TobiasPalludan) [reply lib/blocs/weekplan_bloc.dart]: It has now been back more times than Eminem. However, this time we won't hear from it again.

But I guess they also said that about Eminem.

**Review by MStenkaer — APPROVED** [review]

**Review by jenspetur94 — APPROVED** [review]

**Review by Tekrus — APPROVED** [review]
Since there's no new design part, this is approved. 👍

**Pull request merged**